compiler/rustc_errors/src/lib.rs

```diff
@@ -1995,6 +1995,17 @@ pub fn display_list_with_comma_and<T: std::fmt::Display>(v: &[T]) -> String {
     }
 }
 
+pub fn ordinalize(x: usize) -> String {
```

Comment:
We moved away from ordinalizing in #125105 and #125042 preferring "`#n`" over "`nth`". #125105 has previously removed fn `ordinalize`: https://github.com/rust-lang/rust/pull/125105/files#diff-0e459b4cbb41dd93274d7c68b18a2a3eab7152e9aea107839a7bddff0b4e9195L3031

Reply:
i feel like it's kinda strange that an arbitrary pr gets to set that precedent for how we present diagnostics but ok i guess

Reply:
Yeah, I get that. I don't even remember why this change was done.
We can decide to go back to nth over \#n.
I just want to avoid inconsistency.

Reply:
If y'all decide to go with *nth*, I'll add an item to my backlog to migrate the other places back to it from *\#n*.

Reply:
5134a04eaa32b168cf5998a6ec13199356e2e017 says this:
> Some minor (English only) heroics are performed to print error messages
like "5th rule of macro `m` is never used". The form "rule `#5` of macro
`m` is never used" is just as good and much simpler to implement.

I'm not sure I agree? For me, `#5` is harder to read than `5-th`. And the function that implemented this is like 10 lines, so I'm really unsure what is the win here.

Reply:
Ideally we'd have a style guide for that, but I'm way to exhausted to start a zulip discussion /_=

either way is arguably fine

Reply:
Yeah, seems like a strange thing to justify as a cleanup. But whatever, I really don't want to litigate this; I just want to fix a blatant diagnostics shortcoming. So I'll fix it.

Reply:
cc @nnethercote